libs/libc/locale/lib_gettext.c

```diff
@@ -76,7 +76,7 @@ static FAR const char *g_catname[] =
 static sem_t g_sem = SEM_INITIALIZER(1);
 static FAR struct mofile_s *g_mofile;
 
-#ifdef CONFIG_BUILD_KERNEL
+#if defined(CONFIG_BUILD_KERNEL) || !defined(CONFIG_LIBC_LOCALE)
```

Comment:
@pkarashchenko it doesn't make sense to check CONFIG_LIBC_LOCALE since lib_gettext.c already bypass in Make.defs:
https://github.com/apache/incubator-nuttx/blob/master/libs/libc/locale/Make.defs#L21

Reply:
Oops. I missed that. Will revert this part of the change